ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/KsqlResource.java

```diff
@@ -109,26 +127,38 @@ public KsqlResource(
         Objects.requireNonNull(distributedCmdResponseTimeout, "distributedCmdResponseTimeout");
     this.activenessRegistrar =
         Objects.requireNonNull(activenessRegistrar, "activenessRegistrar");
+    this.injectorFactory = Objects.requireNonNull(injectorFactory, "injectorFactory");
+    this.topicAccessValidator = Objects
+        .requireNonNull(topicAccessValidator, "topicAccessValidator");
+  }
+
+  @Override
+  public void configure(final KsqlConfig config) {
+    if (!config.getKsqlStreamConfigProps().containsKey(StreamsConfig.APPLICATION_SERVER_CONFIG)) {
+      throw new IllegalArgumentException("Need KS application server set");
```

Comment:
nit: change this to include the config name

Reply:
It's not something the client will ever see, or if they did it's nothing they could fix.   If its not set, it's a programming error.